### DISCUSSION TITLE
Beautifying Script Execution logs

### DISCUSSION THREAD

**Discussion by eduardRsult** [burkybang/SuiteAdvanced-Feedback#92] [Ideas]
Format shamelessly copied from [here](https://github.com/burkybang/SuiteAdvanced-Feedback/discussions/78)
I already mentioned this on other places but realise now that this is the best place.

SuiteAdvanced has implemented a nice feature that lets you see script execution notes with beautiful JSON.

This is really cool.

However, someone already does it better : https://chrome.google.com/webstore/detail/netsuite-html-script-note/lipldhgjkmfhamocfcdijcdgjcikcbkk

I would suggest one of the following:

1. remove that feature from SuiteAdvanced
2. have a setting to turn it off so it does not conflict
3. implement it fully. Here's what the other guy does better: not half of the JSON is not beautified ([i made an issue for that since it might just be a bug](https://github.com/burkybang/SuiteAdvanced-Feedback/issues/91)), Instead of linking scriptids to the record type, he links scriptid: internalid to the record instance.   

While your suggestion (on linkedin) to hardcode the url in your script is one way, it takes away the convenience of the other extension. I do think the link to the record type is usefull but less usefull. You could however make the left part to the record type and the right part with the number go to the instance. 

Thanks for making an amazing tool!

**Comment by burkybang** [comment]
This feature is coming in the next release, v27.0.0.

**Comment by eduardRsult** [comment]
Looking forward to it a lot!!!!

**Comment by eduardRsult** [comment]
Awesome

**Comment by burkybang** [comment]
I submitted a new release to the Chrome Web Store. It'll be available whenever Google approves it.

**Reply by burkybang** [reply]
Update: It's been approved.

**Comment by eduardRsult** [comment]
Hi, thanks again, it is quite workable now. 

In detail there are still differences 
secifically no parity when logging an object literal (which is converted to JSON by log.debug())
`recordtype:internalid` is now clickable 
`recordtype: internalid` intenalid is not clickable
{
  "recordtype": "internalid"
}

are both not clickable

{
  "recordtype": intenalid_integer
}

are both not clickable

![Screenshot 2023-08-29 at 13 27 10](https://github.com/burkybang/SuiteAdvanced-Feedback/assets/122311607/71a525b4-c3ea-4e05-8876-0c77fd4657ba)

The other plugin also struggles in some cases (quotationmarks are removed so it is not JSON anymore), so it does not seem to be easy.
![Screenshot 2023-08-29 at 14 06 23](https://github.com/burkybang/SuiteAdvanced-Feedback/assets/122311607/1e607bb1-683b-4aff-8fc0-d3b7284f9564)

**Reply by eduardRsult** [reply]
@burkybang please consider this

**Reply by burkybang** [reply]
@eduardRsult Okay fine. I just completed the feature. It'll be available in the next release.

**Reply by burkybang** [reply]
This is now in the latest version, and it also support object scriptids

**Comment by Fluzzroz** [comment]
I also struggle with this feature. My expectation is that I would log an object of the form {record_type:record_id} and that would be clickable.
I would also be able to combine it with other values for more complete logs. Ex:
`log.debug('my debug log object', {myVar, [myRecord.type]:myRecord.id, another_debug_value:42});`

**Reply by burkybang** [reply]
I just completed this feature. It'll be available in the next release.

**Reply by burkybang** [reply]
This is now in the latest version, and it also support object scriptids

**Comment by burkybang** [comment]
Here's a sneak peak

![image](https://github.com/burkybang/SuiteAdvanced-Feedback/assets/927886/818a9f17-bcaf-479a-b7d1-5eca514fa08f)
